.circleci/config.yml

```diff
@@ -32,11 +32,26 @@ jobs:
         chmod +x ./architect
         ./architect version
 
+    - run: |
+        date +"%Y" > /tmp/year
+        date +"%m" > /tmp/month
+        date +"%d" > /tmp/day
+    - restore_cache:
+        keys:
+        - go-cache-v1-{{ checksum "/tmp/year" }}-{{ checksum "/tmp/month" }}-{{ checksum "/tmp/day" }}
```

Comment:
Can you describe why you are restoring three keys?

Reply:
+1

Reply:
CircleCI uses prefix matching to retrieve the cache, and will try each key in order until it finds one that matches.

So, if we have a cache for the specific day of the build, we'll retrieve that. If we don't (e.g: first build of the day), we'll fallback to the cache from the start of the month (and so on for the first build of the year).

So, we are only restoring one cache, with fallbacks to less fresh caches in case of a miss.

Reply:
Why not just use one key which is independent of date?

Reply:
Caches are immutable, so we'd end up with a cache that is super out of date, and not get any benefit from it.